server/src/main/kotlin/org/javacs/kt/externalsources/KlsURI.kt

```diff
@@ -54,19 +55,20 @@ data class KlsURI(val fileUri: URI, val query: Map<QueryParam, Any>) {
             ?.lastOrNull()
 
     val jarPath: Path
-        get() = Paths.get(fileUri.schemeSpecificPart.split("!")[0])
+        get() = Paths.get(parseURI(fileUri.schemeSpecificPart.split("!")[0]))
```

Comment:
Probably out-of-scope of this PR, but perhaps an even more 'destructured' representation of URIs would be nice, i.e. storing `jarPath` and `innerPath` directly as fields and calling the corresponding parse methods in the constructor.

Reply:
I think it's an interesting idea, but it might take some work, since `KlsURI` is used in a lot of places (and for different purposes). So maybe it should be done in a different PR